tornjak-frontend/src/tables/agents-list-table.tsx

```diff
@@ -7,12 +7,37 @@ import {
     agentsListUpdateFunc
 } from 'redux/actions';
 import Table from './list-table';
+import { AgentsList, AgentsWorkLoadAttestorInfo } from "components/types";
+import { DenormalizedRow } from "carbon-components-react";
+import { RootState } from "redux/reducers";
 
 // AgentListTable takes in 
 // listTableData: agents data to be rendered on table
 // returns agents data inside a carbon component table with specified functions
-class AgentsListTable extends React.Component {
-    constructor(props) {
+
+type AgentsListTableProp = {
+    // dispatches a payload for list of agents with their metadata info as an array of AgentListType and has a return type of void
+    agentsListUpdateFunc: (globalAgentsList: AgentsList[]) => void,
+    // list of available agents as array of AgentsListType or can be undefined if no array present
+    globalAgentsList: AgentsList[] | undefined,
+    // 
+    globalAgentsWorkLoadAttestorInfo: AgentsWorkLoadAttestorInfo[],
+    // the selected server for manager mode 
+    globalServerSelected: string,
+    //
+    data: {
+        key: string,
+        props: { agent: AgentsList }
+    }[] | string | JSX.Element[],
```

Comment:
typing can probably be defined a little more clearer

Reply:
Yes, it can be. That needs to be completely refactored actually. Now we have a new carbon components table so instead of using jsx component we can change the data to a pure array. Created an issue for it.